client/my-sites/customer-home/cards/tasks/site-setup-list/index.jsx

```diff
@@ -131,6 +131,12 @@ const SiteSetupList = ( {
 			( task ) => task.id === CHECKLIST_KNOWN_TASKS.DOMAIN_VERIFIED && ! task.isCompleted
 		).length > 0;
 
+	useEffect( () => {
+		tasks.sort( ( task ) =>
```

Comment:
This is a little dangerous because we're modifying the `tasks` variable that is prop coming to the component.

I'd suggest doing it earlier on the `ConnectedSiteSetupList`. Maybe we could even have a new filter in `WpcomTaskList` to use instead of the `getAll`? We just need to make sure we're not modifying the state.

Reply:
Thanks @renatho! I've used an es6 approach here for sorting and have used a local state for task list, https://github.com/Automattic/wp-calypso/pull/84291/commits/dd809789fb5728f75c3280e4860c8b55a77357d7, LMK if it looks good to you

Reply:
It looks good, but was there a reason to create the additional state?

Instead of the new state and a `useEffect`, wouldn't it be more simple to sort after the `getAll` on the `ConnectedSiteSetupList`? We'd just need to memoize it (From what I can see we have installed in the project, it could be `createSelector` or `memoize`) to avoid the re-rendering. Maybe it was the reason you decided to create the new state?

If you prefer to keep it as it is, I'm fine. Just let me know that I can do a final test to approve. 😉

Reply:
Adding a new function beside `getAll` for sorted output looked like the best way to go here. Didn't require much change in the jsx file, also `getAllSorted` can be used from other places too if needed, with any type of comparator. And the main task list stays untouched. I've made the relevant changes here https://github.com/Automattic/wp-calypso/pull/84291/commits/15dc22f8855e6dbc1a1fb635bb3f6f867d3dc74d. Thanks for all the great suggestions Renatho! LMK if this approach looks good to you.

Reply:
Nice!!! This is looking great!

I just have one last thing about the memoization. Could we memoize the result of the `getAllSorted` based on the `this.tasks`? Otherwise, it will always return a new value to the components, even if there's not change there.

Reply:
Thanks @renatho! I've memoized both the comparator and and sorting function here https://github.com/Automattic/wp-calypso/pull/84291/commits/682fd01fb84394e4bd7fe8b5777fd932340f1fc6.

> Could we memoize the result of the getAllSorted based on the this.tasks

We probably don't need memorize by the `this.tasks` explicitly here because it is already memoized during export-
https://github.com/Automattic/wp-calypso/blob/961cef3ebdda4e56f5093572a14bbfdda6d19574/client/lib/checklist/index.js#L60-L66

And `taskStatuses` is a dependency for the memoize cache key [here](https://github.com/Automattic/wp-calypso/blob/961cef3ebdda4e56f5093572a14bbfdda6d19574/client/lib/checklist/index.js#L63), so when it's updated, we already get a new instance of the sort function

Reply:
Sorry for being a little nitpicky in this part, but I'm thinking that it's important since it will impact other parts of the application not related to Sensei too. :)

> memoized both the comparator

Do you think we really need to memoize the `unverifiedEmailTaskComparator`? I think it's just memoizing the internal function based on the `isEmailUnverified` argument, right? I think this memoization wouldn't be needed with the `getAllSorted` memoized, right?

> and sorting function

I think we should send the second argument to the `memoize` to make the memoization based on the `this.tasks`? Otherwise, I think it will memoize based on the `comparator`.

Reply:
Hey! Just to document here, based on what we talked in Slack, I'm approving this. That we want to memoize that comparator anyway, to re-run the selector in case it's changed. Thank you for working on this, Imran!